src/main/java/gregtech/api/pipenet/longdist/BlockLongDistancePipe.java

```diff
@@ -0,0 +1,105 @@
+        this.pipeType = pipeType;
+        setTranslationKey("long_distance_" + pipeType.getName() + "_pipeline");
+        setCreativeTab(GregTechAPI.TAB_GREGTECH);
+    }
+
+    @Override
+    public void onBlockPlacedBy(@Nonnull World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state, @Nonnull EntityLivingBase placer, @Nonnull ItemStack stack) {
+        super.onBlockPlacedBy(worldIn, pos, state, placer, stack);
+        if (worldIn.isRemote) return;
+        List<LongDistanceNetwork> networks = new ArrayList<>();
+        BlockPos.PooledMutableBlockPos offsetPos = BlockPos.PooledMutableBlockPos.retain();
+        for (EnumFacing facing : EnumFacing.VALUES) {
+            offsetPos.setPos(pos).move(facing);
+            LongDistanceNetwork network = LongDistanceNetwork.get(worldIn, offsetPos);
+            if (network != null && pipeType == network.getPipeType()) {
+                MetaTileEntityLongDistanceEndpoint endpoint = MetaTileEntityLongDistanceEndpoint.tryGet(worldIn, offsetPos);
+                // only count the network as connected if it's not an endpoint or the endpoints input or output face is connected
+                if (endpoint == null || endpoint.getFrontFacing().getAxis() == facing.getAxis()) {
+                    networks.add(network);
+                }
+            }
+        }
+        offsetPos.release();
+        if (networks.isEmpty()) {
+            // create network
+            LongDistanceNetwork network = this.pipeType.createNetwork(worldIn);
+            network.onPlacePipe(pos);
+        } else if (networks.size() == 1) {
+            // add to connected network
+            networks.get(0).onPlacePipe(pos);
+        } else {
+            // merge all connected networks together
+            LongDistanceNetwork main = networks.get(0);
+            main.onPlacePipe(pos);
+            networks.remove(0);
+            for (LongDistanceNetwork network : networks) {
+                main.mergePipeNet(network);
+            }
+        }
+    }
+
+    @Override
+    public void breakBlock(@Nonnull World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state) {
+        super.breakBlock(worldIn, pos, state);
+        if (worldIn.isRemote) return;
+        LongDistanceNetwork network = LongDistanceNetwork.get(worldIn, pos);
+        network.onRemovePipe(pos);
+    }
+
+    @Override
+    public void getSubBlocks(@Nonnull CreativeTabs itemIn, @Nonnull NonNullList<ItemStack> items) {
+        if (itemIn == GregTechAPI.TAB_GREGTECH) {
+            items.add(new ItemStack(this));
+        }
+    }
+
+    public enum Type implements IStringSerializable {
+        FULL,
+        X,
+        Y,
+        Z,
+        X_PLUS,
+        Y_PLUS,
+        Z_PLUS,
+        X_MINUS,
+        Y_MINUS,
+        Z_MINUS,
+        X_BOTH,
+        Y_BOTH,
+        Z_BOTH;
+
+        @Nonnull
+        @Override
+        public String getName() {
+            return name().toLowerCase();
+        }
+    }
```

Comment:
Dead code

If this was for the proposed idea of a more complex model for the pipe blocks, I personally think we shouldn't do that. While it would look nice, the purpose of long distance pipes is to transport items/fluids obscene distances, and forcing complex models for these blocks adds unnecessary rendering strain in my opinion. And the idea of a config option for a more advanced model for these I feel is unnecessary, I think just the blocks as they are is fine